gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -22,63 +24,88 @@ def __init__(self, service):
                 ["read", "write", "notify"], service)
         self.add_descriptor(WifiConnectDescriptor(self))
         self.add_descriptor(OpaqueStructureDescriptor(self))
-        self.wifi_status = ""
 
-    def WiFiConnectCallback(self):
+        # Credential to connect to a WiFi
+        self.wifi_service = ""
+        self.wifi_password = ""
+
+    def check_wifi_status(self):
+        # Check the current wi-fi connection status
+        logger.debug('Check WiFi Connect')
+        nm_state = str(nmcli_custom.device.show('wlan0')['GENERAL.STATE'].split(" ")[0])
+
+        # Convert the network manager device state into wifi status response
+        if nm_state == constants.NM_DEVICE_STATE_ACTIVATED:
```

Comment:
I think a combination of this style with the original would be best. Meaning, using a dict that maps  network manager states (`NM_`)  to wifi statuses (`WIFI_`).

Reply:
I agree.